components/server/src/auth/resource-access.ts

```diff
@@ -176,11 +169,16 @@ export class OwnerResourceGuard implements ResourceAccessGuard {
                         // Only owners can update or delete a team.
                         return resource.members.some(m => m.userId === this.userId && m.role === "owner");
                 }
-                case "workspaceLog":
-                    return resource.subject.ownerId === this.userId;
         }
     }
 
+    protected hasAccessToWorkspace(workspace: Workspace, teamMembers?: TeamMemberInfo[]): boolean {
+        return workspace.ownerId === this.userId
+            || (workspace.type === 'prebuild'
+                && (!workspace.projectId /* legacy mode: prebuilds without projects are accessible by everyone */
```

Comment:
As discussed on Slack, allowing everyone to access prebuilds that aren't associated to a project might be a bit dangerous (e.g. what if a user can access the prebuild logs of a private repository they shouldn't have access to?)

Maybe it could be a good idea to keep the `canAccessHeadlessLogs` method and use it here in legacy mode? 🤷‍♂️